client/src/backoffice/QrPage.tsx

```diff
@@ -0,0 +1,213 @@
+import React, { FC, useState, useEffect, ChangeEvent } from 'react';
+
+/* Libraries */
+import ReactPaginate from 'react-paginate';
+
+/* Components */
+import { Loading } from '../components/Loading';
+
+/* Helpers */
+import { QrCode, getQrCodes, getEvents, PoapEvent } from '../api';
+import { reduceAddress } from '../lib/helpers';
+import { etherscanLinks } from '../lib/constants';
+
+/* Assets */
+import checked from '../images/checked.svg';
+import error from '../images/error.svg';
+
+/* Typings */
+import { Value } from '../types';
+
+const PAGE_SIZE = 10;
+
+type PaginateAction = {
+  selected: number;
+};
+
+export interface QrCode {
+  id: number;
+  qr_hash: string;
+  claimed: boolean;
+  tx_hash: string;
+  event_id: number;
+  event: PoapEvent;
+}
+
+const QrPage: FC = () => {
+  const [page, setPage] = useState<number>(0);
+  const [total, setTotal] = useState<number>(0);
+  const [isFetchingQrCodes, setIsFetchingQrCodes] = useState<null | boolean>(null);
+  const [qrCodes, setQrCodes] = useState<null | QrCode[]>(null);
+  const [claimStatus, setClaimStatus] = useState<string[]>([]);
+  const [selectedEvent, setSelectedEvent] = useState<number | undefined>(undefined);
+  const [events, setEvents] = useState<PoapEvent[]>([]);
+
+  useEffect(() => {
+    fetchEvents();
+  }, []);
+
+  useEffect(() => {
+    fetchQrCodes();
+  }, [page]);
+
+  useEffect(() => {
+    page !== 0 ? setPage(0) : fetchQrCodes();
+  }, [selectedEvent, claimStatus]);
+
+  const fetchEvents = async () => {
+    const events = await getEvents();
+    setEvents(events);
+  };
+
+  const fetchQrCodes = async () => {
+    setIsFetchingQrCodes(true);
+
+    let status = undefined;
+    if (claimStatus.length === 1) status = claimStatus[0] === 'claimed';
+
+    let event_id = undefined;
+    if (selectedEvent !== undefined) event_id = selectedEvent > -1 ? selectedEvent : undefined;
+
+    try {
+      const response = await getQrCodes(PAGE_SIZE, page * PAGE_SIZE, status, event_id);
+      if (!response) return;
+      setQrCodes(response.codes);
+      setTotal(response.total);
+    } catch (e) {
+      console.log(e);
```

Comment:
We're not displaying any error messages to the user/admin would be great to have a `Toast` displaying async errors i.e `We couldn't get new notifications`

Reply:
we are now displaying the error message sent from the backend team

Reply:
we are now displaying the error message sent from the backend team

Reply:
Great now the user has at least some kind of feedback if something wrong happened.